mne/channels/layout.py

```diff
@@ -375,6 +375,10 @@ def find_layout(info, ch_type=None, exclude='bads'):
     exclude : list of string | str
         List of channels to exclude. If empty do not exclude any.
         If 'bads', exclude channels in info['bads'] (default).
+    on_failure : None | 'generate'
+        What to do if no layout is found. If None (default), None is returned.
+        If 'generate', a custom layout is generated from all the channel
+        locations.
```

Comment:
the current behavior is that find_layout tries to generate a layout if it can (included making one for EEG) and returns None if it fails. So the logic is "generate" by default. Now I don't get why we need this on_failure option.